mobile/src/home-view.js

```diff
@@ -139,15 +139,28 @@ export default class HomeView extends Component {
   }
 
   saveItem = () => {
-    fbc.database.public.userRef('items').push(this.state.currentItem)
-    .then(() => {
-      setTimeout(() => {
-        this.changeView("home")
-        this.setState({currentItem: {}, itemStage: 0})
-        }
-        ,250)
-    })
-    .catch(error => this.setState({questionError: "Retry"}))
+    if (this.state.currentItem.id) {
+      fbc.database.public.userRef('items').child(this.state.currentItem.id).update(this.state.currentItem)
+      .then(() => {
```

Comment:
Both cases appear to do the same then/catch with the Promise.  Just branch the logic to do an update vs. push and then `.then`/`.catch` the resulting Promise rather than repeating the then/catch

Reply:
```javascript
const itemsRef = fbc.database.public.userRef('items')
const update = this.state.currentItem.id
  ? itemsRef.child(this.state.currentItem.id).update(this.state.currentItem)
  : itemsRef.push(this.state.currentItem)
update.then(() => {
  ...
}).catch(...)
```